### ISSUE TITLE
Reek 2 final spurt: release notes and things left to do

### ISSUE THREAD

**Issue by troessner** [troessner/reek#365]
Hi guys!

I'd like to release reek 2 after we merged all the current pull requests that are mergeable ( #358 has external dependencies after all) and I'd like to make a fuss about it, at least a little one.:)

So basically I'd just write a fancy release note, publish it here on github or on my blog and just link to it here (not sure if github is the right place for this) and then send it that to the fantastic ruby newsletter in the hope to get mentioned. Maybe other publications as well (I don't know of any others, but maybe you do?).

**So what would you guys like to see in the release notes?**

Looking at the [CHANGELOG](https://github.com/troessner/reek/blob/master/CHANGELOG) there are so many features that have been implemented since 1.0. Pretty hard to do a summary there.:)

**From my point of view:**

**Support:**

Parse with the parse gem allowing us to support all major ruby versions:
- 1.9.3
- 2.0
- 2.1
- 2.2. (that one obviously depending on #358 )

**New smells:**

- ModuleInitialize smell
- Prima Donna Method smell

**CLI**

Completely revised, see #254 

**Configuration:**

- Revise configuration handling to use one configuration file only (and try to look it up if you can't find it)
- Add support for detecting specific smells
- Smell detectors can be configured or disabled in code comments

**Output:**

- Add ultra verbose warning formatter.
- Add color to reek's output
- Added line numbers by default when you run the reek command
- Allow sorting by issue count
- Added total warning count
- yaml and html output

**rspec-matcher**

Improved matcher, see #319 

**Bugfixes:**

Not sure about this point. We had so many that this seems kind of pointless (especially since this is kind of irrelevant to "external observers")

------------------------------

What do you guys think of that?
Any other suggestions?
What would you like to see in there?

**Comment by mvz** [comment]
I like the idea of making at least a little fuss about this. Reek is a great tool and everyone should use it.

One thing we should give attention to is any breaking changes in 2.0. We've been moderately careful in not breaking stuff during the 1.x series, although some breaking changes snuck in. It may be good to finalize those (e.g., the yaml output), but now is also a good time to remove anything we consider deprecated. The config_files option from the rake task comes to mind, and maybe some other options there.

**Comment by troessner** [comment]
>> It may be good to finalize those (e.g., the yaml output),

I agree - do you have a list of things in mind that makes sense to you? If yes, let's collect it here (i also renamed this issue so that it's less confusing).
How about #254 ?

**Comment by troessner** [comment]
Let's tag all issues that are left for reek-2 / will be in reek-2 with the label..."reek-2" (just created it).
Oh, and I'll also submit this to hackernews once it's done.:)

**Comment by mvz** [comment]
I am getting to know GitHub's milestones, so I created a 2.0 milestone. Not sure how well that works yet. Maybe we should also create separate branches? Perhaps we can branch off a 1-6-stable branch, or alternatively, a 2-0 branch to be merged into master. That way, we don't have to leave the 2.0 pull requests hanging around.

**Comment by troessner** [comment]
>> I am getting to know GitHub's milestones, so I created a 2.0 milestone. 

Never used that, but from what I can read here: https://guides.github.com/features/issues/ this makes absolutely sense.

>> Maybe we should also create separate branches? 

Yeah, so far we kind of elegantly evaded this set up for simplicity.:D

>> Perhaps we can branch off a 1-6-stable branch, or alternatively, a 2-0 branch to be merged into master

Yes, let's go with a 1.6 branch, I believe that's the right way as well.
I just created this branch.
This means we can merge the reek-2 PR's and just never again touch the 1.6. branch unless for hotfixes.:)

**Comment by troessner** [comment]
Addendum: I called that branch "v1.6", not just "1.6"

**Comment by mvz** [comment]
Hm ... that might be a little confusing with all those `vx.y` tags we have.

**Comment by troessner** [comment]
Hmmm, good point. Just "1.6." would probably also be a little confusing, don't you think?
How about "stable-1.6"?

**Comment by mvz** [comment]
stable-1.6 sounds good to me.

**Comment by troessner** [comment]
Done.

**Comment by troessner** [comment]
Ok, so let me quickly summarize our status here:

1.) We fix #326, #319, #254 
2.) We release 2.0
3.) I'll write an article based on the draft above and publish it
4.) I'll try to spread the word (read: the article) via Ruby NL and hacker news.

**Comment by troessner** [comment]
@mvz how about we split up the 2 remaining  tasks?
How about i take #319 and you take #254 ?

**Comment by mvz** [comment]
Sure, sounds like a plan.

**Comment by troessner** [comment]
@mvz now with everything marked reek-2 out of the way let's go for the 2 release!

I prepared a first draft  for the article i'd submit for hacker news and the ruby newsletter: https://gist.github.com/troessner/c9050bebd94e2c61de06

Let me know what you think!

**Comment by troessner** [comment]
Aaaaaaand v2.0.0 is out!

-> https://github.com/troessner/reek/commit/2aef274c20817e754ca0ff3d319a4e6d8abd9905

Now let's get this article going and then announce this to the world.:)

**Comment by mvz** [comment]
Updating reek ... now!
